release-to-candidate/action.yaml

```diff
@@ -43,23 +49,56 @@ runs:
         git config --global user.email "github-actions@github.com"
         git config --global user.name "Github Actions"
 
-        # Install moreutils so we have access to sponge
-        sudo apt-get update; sudo apt-get install -y moreutils
+    - name: Find the snapcraft.yaml path
+      id: yaml-path
+      shell: bash
+      run : |
+        if [[ -n "${{ inputs.snapcraft-yaml-path }}"; then
+          yaml_path="${{ inputs.snapcraft-yaml-path }}"
+        else
+          snapcraft_yaml_paths=(
+            "snap/snapcraft.yaml"
+            "snapcraft.yaml"
+            "build-aux/snap/snapcraft.yaml"
+            ".snapcraft.yaml"
+          )
+          
+          for file in "${snapcraft_yaml_paths[@]}"; do
+              if [[ -f "$file" ]]; then
+                  yaml_path="$file"
+              fi
+          done
+        fi
+        if [[ ! -n "${yaml_path}" ]]; then
+           echo "No snapcraft.yaml found" >2
+           exit 1
+        fi
+        echo "yaml_path=${yaml_path}" >> "$GITHUB_OUTPUT"
+
 
     - name: Build the snap (${{ inputs.architecture }})
       id: build
       shell: bash
+      with:
+        yaml_path: ${{ steps.yaml-path.outputs.yaml_path }}
       env:
-        name: ${{ github.event.repository.name }}
+        name: ${{ inputs.snap-name || github.event.repository.name }}
         arch: ${{ inputs.architecture }}
       run : |
         # Remove the architecture definition from the snapcraft.yaml due to:
         # https://bugs.launchpad.net/snapcraft/+bug/1885150
-        cat snap/snapcraft.yaml | yq 'del(.architectures)' | sponge snap/snapcraft.yaml
+        yq -i 'del(.architectures)' "$yaml_path"
 
         snapcraft remote-build --launchpad-accept-public-upload --build-for="${arch}"
 
-        version="$(cat snap/snapcraft.yaml | yq -r '.version')"
+        version="$(yq -r '.version' "$yaml_path")"
+        if [[ ! -e ${name}_${version}_${arch}.snap ]]; then
+            echo "Could not find ${name}_${version}_${arch}.snap"
+            cat ${name}_${arch}.txt
```

Comment:
What is this for? Either way it needs to be quoted :)

Reply:
It's the build log, #3